circuitClient.js

```diff
@@ -122,6 +124,68 @@ class CircuitClient {
       }]);
     });
   }
+  
+  async setMyStatusMessage(statusMessage) {
+    return this.client.setStatusMessage(statusMessage)
+    .catch(console.error);
+  }
+
+  getMyStatusMessage() {
+    return this.client.getStatusMessage()
+    .catch(console.error);
+  }
+
+  async setPresenceAvailable() {
+    const statusMsg = await this.client.getStatusMessage();
+
+    return this.client.setPresence({
+      state: Circuit.Enums.PresenceState.AVAILABLE,
+      statusMessage: statusMsg
+    }).catch(console.error);
+  }
+  
+  async setPresenceDnd(untilTime, duration) {
+
+    const statusMsg = await this.client.getStatusMessage();
+
+    if (untilTime !== '') {
```

Comment:
might be better to use a switch here alongand try to only return something at the end. I.e  set a blank status message and have it run through the switch statement and assign it (along with the calculated time) , then call the setPresence based on the logic in your switch statement